src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -196,10 +196,16 @@ private static DependencyGraphSpec ToDependencyGraphSpec(ProjectNames projectNam
                     .Any(tfm => tfm.Version >= Version20);
                 var toolFramework = isNetCore20 ? CommonFrameworks.NetCoreApp20 : CommonFrameworks.NetCoreApp10;
 
+                var packagesPath = GetRestoreProjectPath(projectRestoreInfo.TargetFrameworks);
+                var fallbackFolders = GetRestoreFallbackFolders(projectRestoreInfo.TargetFrameworks).ToList();
+                var sources = GetRestoreSources(projectRestoreInfo.TargetFrameworks)
+                             .Select(e => new PackageSource(e))
+                             .ToList();
+
                 projectRestoreInfo
                     .ToolReferences
                     .Cast<IVsReferenceItem>()
-                    .Select(r => ToToolPackageSpec(projectNames, r, toolFramework))
+                    .Select(r => ToToolPackageSpec(projectNames, r, toolFramework, packagesPath, fallbackFolders, sources, null))
                     .ToList()
```

Comment:
The Enumerable Type does not have a ForEach method. 
I'll add an extension method so we don't do the iteration twice (once toList, once forEach)